File: data_baseline/openrouter__qwen__qwen3-coder-plus/java/rep1/TodoService.java
import java.util.*;
import java.util.concurrent.ConcurrentHashMap;
import java.time.Instant;
import java.time.ZoneOffset;
import java.time.format.DateTimeFormatter;
import java.time.format.DateTimeFormatterBuilder;
import java.time.temporal.ChronoField;

public class TodoService {
    private final Map<Integer, Todo> todosById = new ConcurrentHashMap<>();
    private final Map<Integer, List<Todo>> userTodosMap = new ConcurrentHashMap<>(); // userId -> todos
    private Integer nextTodoId = 1;
    
    private static final DateTimeFormatter FORMATTER = new DateTimeFormatterBuilder()
        .appendPattern("yyyy-MM-dd'T'HH:mm:ss'Z'")
        .toFormatter();
    
    public synchronized int getNextTodoId() {
        return nextTodoId++;
    }
    
    public Todo createTodo(int userId, String title, String description) {
        Todo todo = new Todo();
        todo.id = getNextTodoId();
        todo.title = title;
        todo.description = description != null ? description : "";
        todo.completed = false;
        Instant now = Instant.now();
        todo.created_at = FORMATTER.format(now.atOffset(ZoneOffset.UTC));
        todo.updated_at = FORMATTER.format(now.atOffset(ZoneOffset.UTC));
        
        todosById.put(todo.id, todo);
        
        // Register this todo under the user who owns it
        userTodosMap.computeIfAbsent(userId, k -> new ArrayList<>()).add(todo);
        
        return todo;
    }
    
    public Todo getTodoById(int todoId) {
        return todosById.get(todoId);
    }
    
    public List<Todo> getTodosForUser(int userId) {
        List<Todo> userTodos = userTodosMap.getOrDefault(userId, new ArrayList<>());
        // Sort by ID ascending
        userTodos.sort(Comparator.comparingInt(t -> t.id));
        return new ArrayList<>(userTodos); // Return a copy for thread safety
    }
    
    public Todo updateTodo(int todoId, String title, String description, Boolean completed) {
        Todo todo = todosById.get(todoId);
        if (todo != null) {
            if (title != null) {
                if (title.trim().isEmpty()) {
                    throw new IllegalArgumentException("Title cannot be empty");
                }
                todo.title = title;
            }
            if (description != null) {
                todo.description = description;
            }
            if (completed != null) {
                todo.completed = completed;
            }
            
            // Update the timestamp
            Instant now = Instant.now();
            todo.updated_at = FORMATTER.format(now.atOffset(ZoneOffset.UTC));
        }
        return todo;
    }
    
    public boolean deleteTodo(int todoId) {
        Todo todo = todosById.remove(todoId);
        if (todo != null) {
            // Remove from user's todo list
            for (List<Todo> userList : userTodosMap.values()) {
                userList.removeIf(t -> t.id == todoId);
            }
            return true;
        }
        return false;
    }
    
    public boolean isTodoOwnedByUser(int todoId, int userId) {
        List<Todo> userTodos = userTodosMap.get(userId);
        if (userTodos == null) return false;
        return userTodos.stream().anyMatch(todo -> todo.id == todoId);
    }
}

File: data_baseline/openrouter__qwen__qwen3-coder-plus/scala/rep1/TodoService.scala
import scala.collection.mutable
import java.util.concurrent.atomic.AtomicInteger

class TodoService {
  private val todos = mutable.Map[Int, Todo]()
  private val todoCounter = new AtomicInteger(1)

  // Helper methods for managing todos
  def createTodo(title: String, description: String, userId: Int): Todo = synchronized {
    val id = todoCounter.getAndIncrement()
    val now = DateTimeHelper.nowAsString()
    val todo = Todo(
      id = id,
      title = title,
      description = description,
      completed = false,
      createdAt = now,
      updatedAt = now,
      userId = userId
    )
    todos.put(id, todo)
    todo
  }

  def getTodosByUserId(userId: Int): List[Todo] = {
    todos.values.filter(_.userId == userId).toList.sortBy(_.id)
  }

  def getTodo(todoId: Int, userId: Int): Option[Todo] = {
    todos.get(todoId).filter(_.userId == userId)
  }

  def updateTodo(todoId: Int, userId: Int, title: Option[String], description: Option[String], completed: Option[Boolean]): Option[Todo] = synchronized {
    todos.get(todoId) match {
      case Some(todo) if todo.userId == userId =>
        val newTitle = title.getOrElse(todo.title)
        val newDescription = description.getOrElse(todo.description)
        val newCompleted = completed.getOrElse(todo.completed)
        val updatedTodo = todo.copy(
          title = newTitle,
          description = newDescription,
          completed = newCompleted,
          updatedAt = DateTimeHelper.nowAsString()
        )
        todos.update(todoId, updatedTodo)
        Some(updatedTodo)
      case _ => None
    }
  }

  def deleteTodo(todoId: Int, userId: Int): Boolean = synchronized {
    todos.get(todoId) match {
      case Some(todo) if todo.userId == userId =>
        todos.remove(todoId)
        true
      case _ => false
    }
  }
}